antarest/study/service.py

```diff
@@ -1894,7 +1894,7 @@ def unarchive(self, uuid: str) -> str:
 
         def unarchive_task(notifier: ITaskNotifier) -> TaskResult:
             study_to_archive = self.get_study(uuid)
-            self.storage_service.raw_study_service.unarchive(study_to_archive)
+            self.storage_service.raw_study_service.unarchive(cast(RawStudy, study_to_archive))
```

Comment:
Ok here we are sure it's a RawStudy (checked above).
However, I would still prefer to have a small method like `assert_raw` which actually check the type, otherwise someone could remove the check above and we won't be covered here anymore.
```python
def assert_raw(study: Study) -> RawStudy:
    if not isinstance(study, RawStudy):
        raise <something>
    return study
```